### ISSUE TITLE
Floating point __str__

### ISSUE THREAD

**Issue by anntzer** [numpy/numpy#6908]
Related to (but different from) #6136: `float64.__str__` can truncate different floats to the same string, even when `printoptions` is set to a high enough precision:

```
In [12]: np.set_printoptions(precision=1000)

In [13]: x = np.float64(0.20000000000000004)

In [14]: x > np.float64(.2)
Out[14]: True

In [15]: str(x)
Out[15]: '0.2' # oops
```

Note that `repr(x)` correctly gives `'0.20000000000000004'`.

**Comment by gfyoung** [comment]
Is it me, or are the print options <b>only respected</b> when we want to print arrays?  Here's another example of "bad" behavior with scalars:
```
>>> import numpy as np
>>> x = np.float64(0.12345678901234567890123456789)
>>> m = np.array([x])
>>>
>>> str(x) # default precision is 8 decimal places
'0.123456789012' # oops
>>> str(m) # printing an array containing 'x' respects the precision
'[ 0.12345679]'
```

**Comment by sam09** [comment]
Is anyone working on this issue?
If not i would like to give it a go

**Comment by gfyoung** [comment]
I don't know of anyone who is.  Generally, IIUC the etiquette is that unless someone is really talking about it <b>or</b> has already opened a PR to fix the issue, it is completely up for grabs by anyone.  So go for it!  Good luck!

**Comment by anntzer** [comment]
Please consider fixing #6136 at the same time :)

**Comment by sam09** [comment]
It seems the ```printoptions``` are set in ```numpy/core/arrayprint.py```. It has functions defined for printing an array but not single numpy values. Is this behavior intended? I mean is printoptions set to work only for arrays? [Here](http://stackoverflow.com/questions/12956333/printing-numpy-float64-with-full-precision) is link to an answered question on stackoverflow. Please tell me if I am missing something

**Comment by gfyoung** [comment]
Judging from the labeling by @rgommers , I suspect not.  I think you may very well have to add a good amount of new code in order to enforce printing for numbers.

**Comment by sam09** [comment]
I think the float values are also printed by array2string  function defined in  ``numpy/core/arrayprint.py``.
[This](https://github.com/numpy/numpy/blob/master/numpy/core/arrayprint.py#L437) is where I think it happens

**Comment by gfyoung** [comment]
You still have to cast ```a``` as an ```array``` object.  It's just one without a shape, so the function still does not apply to pure literals like ```a = 5``` for example.

**Comment by NeilGirdhar** [comment]
I just ran into this issue.  printoptions should be respected by all three:
```
In [21]: np.set_printoptions(precision=3)

In [22]: print(np.ones(()) * (0.123456789123))
0.123456789123

In [23]: print(np.float32(0.123456789123))
0.123457

In [24]: print(np.ones((1,)) * (0.123456789123))
[ 0.123]
```

**Comment by ahaldane** [comment]
Closed by #9941
